src/main/java/subway/domain/Distance.java

```diff
@@ -0,0 +1,39 @@
+package subway.domain;
+
+import java.util.Objects;
+
+public class Distance {
+    private final int distance;
+
+    public Distance(final int distance) {
+        if (distance <= 0) {
+            throw new IllegalArgumentException("구간의 거리는 0보다 커야 합니다.");
```

Comment:
간단한 미션이나 토이프로젝트에선 자주 통용되는 예외인 IllegalArgumentException 등을 사용하는데요, 실제 코드에선 커스텀 예외 (RuntimeException을 상속하여 서비스에서 발생하는 예외를 위한 예외)를 사용하고, 해당 예외를 위한 Handler를 만들어 쓰는 경우가 많습니다. 

IllegalArgumentException가 개발 전반적으로 많이 쓰이는 코드다보니 라이브러리에서 자주 올라오기 때문에 IllegalArgumentException에 대해 Handling을 하면 코드에서 쓰는 라이브러리 예외 정보가 클라이언트에게 노출되고 취약점으로 이어질수 있기 때문인데요

미션에서 다루는 핵심 주제는 아니니 참고만 해주세요.

Reply:
그리고 예외 메시지는 항상 풍부하게 작성해주는 편이 좋습니다. 실제로 입력된 distance가 어떤 값인지 노출하도록 해줄까요?
```suggestion
            throw new IllegalArgumentException("구간의 거리는 0보다 커야 합니다. 입력된 값 : " + distance);
```

Reply:
예외처리 더 자세하게 할 수 있도록 노력해 보겠습니다😊